src/store/index.ts

```diff
@@ -305,6 +305,66 @@ export default (api: Kuma) => {
       SET_SUPPORTED_VERSIONS_FAILED: (state, value) => (state.supportedVersionsFailed = value),
     },
     actions: {
+      // bootstrap app
+
+      async bootstrap ({ commit, dispatch, getters }, routeMesh) {
+        // check the API status before we do anything else
+        await dispatch('getStatus')
+        // only dispatch these actions if the API is online
+        if (getters.getStatus === 'OK') {
+          // set the current environment
+          commit('SET_ENVIRONMENT', localStorage.getItem('kumaEnv'))
+
+          // fetch the mesh list
+          dispatch('fetchMeshList')
+          // fetch the tagline
+          dispatch('getTagline')
+          // fetch the config
+          dispatch('getConfig')
+
+          // fetch the version and store it in localStorage
+          dispatch('getVersion')
+            .then(() => {
+              const newVersion = getters.getVersion
+              const lsVersion = localStorage.getItem('kumaVersion') || null
+              // if the version stored in the browser is different than the
+              // version running, update the version in localStorage, and
+              // reload the page
+
+              if (lsVersion !== newVersion) {
+                // reload the app
+                window.location.reload()
+                // update the version in localStorage
+                localStorage.setItem('kumaVersion', newVersion)
+              } else {
+                store.commit('SET_GLOBAL_LOADING', { globalLoading: false })
+              }
+            })
```

Comment:
As the function is async, what do you think about instead of using promises' `.then` method here to use `await`?

Reply:
I thought about it but I decided to do it, later on, to do not change logic everywhere when doing cleanup.
Also when working with `async/await` we need to remember that it is blocking the flow of execution of the function, so if we just change everything to use `async/await` then every call needs to wait for the previous call to be executed.
The other option might be to aggregate all `Promises` and `await Promise.allSettled` to make sure we do not delay the resolution of it. But as I said, I will think about it later on.